sdk/iot/hub/src/az_iot_hub_client.c

```diff
@@ -140,3 +142,48 @@ AZ_NODISCARD az_result az_iot_hub_client_properties_append(
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_iot_hub_client_properties_find(
+    az_iot_hub_client_properties* properties,
+    az_span name,
+    az_span* out_value)
+{
+  AZ_PRECONDITION_NOT_NULL(properties);
+  AZ_PRECONDITION_VALID_SPAN(name, 1, false);
+  AZ_PRECONDITION_NOT_NULL(out_value);
+
+  az_span remaining = properties->_internal.properties;
+
+  int32_t index = 0;
+  // Keep looking in case the passed name is a substring of a name in props
+  while (az_span_length(remaining) != 0)
+  {
+    index = az_span_find(remaining, name);
+    if (index != -1)
+    {
+      az_span value;
+      az_span found_name;
+      remaining = az_span_slice(remaining, index, -1);
+      found_name = az_span_token(remaining, hub_client_param_equals_span, &value);
```

Comment:
If the property is "key_one=value_one", "value_one" would be found with name "ey_one". Check for length of `found_name` is not restrict.

Reply:
Ahh you’re right. Had that in but removed on a rework. Thanks will update.

Reply:
Added this:
https://github.com/Azure/azure-sdk-for-c/blob/4677584b73519bbe90c7abb4de91dd92df458ab7/sdk/iot/hub/src/az_iot_hub_client.c#L180-L184
And added tests which test if a previous value matches what you are looking for, ie searching for `key` in this:
https://github.com/Azure/azure-sdk-for-c/blob/4677584b73519bbe90c7abb4de91dd92df458ab7/sdk/iot/hub/tests/cmocka/test_az_iot_hub_client.c#L49
both in the success case and fail case, and the edge case when the value is the last value (where this addition makes sense as there is no remaining chars to search.)